crates/rust-analyzer/src/semantic_tokens.rs

```diff
@@ -45,15 +45,16 @@ define_semantic_token_types![
     (BRACKET, "bracket"),
     (BUILTIN_TYPE, "builtinType"),
     (CHAR_LITERAL, "characterLiteral"),
-    (COMMA, "comma"),
     (COLON, "colon"),
+    (COMMA, "comma"),
+    (CONST_PARAMETER, "constParameter"),
     (DOT, "dot"),
     (ESCAPE_SEQUENCE, "escapeSequence"),
     (FORMAT_SPECIFIER, "formatSpecifier"),
     (GENERIC, "generic"),
-    (CONST_PARAMETER, "constParameter"),
-    (LIFETIME, "lifetime"),
+    (INTRA_DOC_LINK, "intraDocLink"),
```

Comment:
Do we need a separate token *type* here? I think that we might want to set the type to the type of the pointed to item, and set modifier to "doc link". That is, treat links (the part that is a rust path) as small snippets of injected rust code.

Reply:
Oh that's a nice idea actually